.azure-pipelines/templates/build.yml

```diff
@@ -0,0 +1,87 @@
+parameters:
+- name: py_version
+  type: string
+  default: ''
+- name: image
+  type: string
+  default: ''
+- name: ccache
+  type: boolean
+  default: false
+- name: OSX_VERSION
+  type: string
+  default: ''
+- name: publish
+  type: string
+  default: ''
+
+jobs:
+  - job:
+    timeoutInMinutes: 120
+    displayName: ${{ format('{0}-py{1}', parameters.image, parameters.py_version) }}
+    pool:
+      vmImage: ${{ parameters.image }}
+    variables:
+      - name: OSX_VERSION
+        value: ${{ parameters.OSX_VERSION }}
+
+    steps:
+      - checkout: self
+        submodules: true
+
+      - ${{ if eq(parameters.ccache, true) }}:
+        - template: ccache.yml
+
+      # Note that using set -ex when setting vso variables messes up paths
+      - ${{ if contains(parameters.image, 'windows') }}:
+        - bash: |
+            echo "##vso[task.prependpath]$CONDA\Scripts"
+            echo "##vso[task.setvariable variable=file_extension]bat"
+          displayName: 'Configure Conda path'
+      - ${{ if not(contains(parameters.image, 'windows')) }}:
+        - bash: |
+            echo "##vso[task.prependpath]$CONDA/bin"
+            echo "##vso[task.setvariable variable=file_extension]sh"
+          displayName: 'Configure Conda path'
+      - bash: echo "##vso[task.setvariable variable=conda_dir]$CONDA"
+        displayName: 'Configure Conda dir'
+      - ${{ if not(eq(length(parameters.OSX_VERSION), 0)) }}:
+        - bash: |
+            set -ex
+            sudo chown -R $USER $CONDA
+          displayName: 'Take ownership of Conda installation'
+
+      - bash: |
+          set -ex
+          conda --version
+          conda install --yes conda-build
+          conda config --set always_yes yes --set changeps1 no
+          # Specify python version in env file
+          sed 's/- python$/- python=${{ parameters.py_version }}/g' scipp-developer-minimal.yml > temp-env.yml
+          conda env create -f temp-env.yml
+        displayName: 'Create conda environment'
+
+      - bash: |
+          set -ex
+          source activate scipp-developer
+          tools/make_and_install.$FILE_EXTENSION
+        displayName: 'Build and run C++ tests'
```

Comment:
Isn't this also "installing"?

Reply:
Yes. It depends how long we want the names of the tasks to be.

I also think the name `Python tests and conda build` for the next step is not great either.
Technically it should be `conda build and python tests`.
But i thought it's useful to have `Python tests` at the start of the name because it's actually what you are looking for when you are checking your build, and if you have a narrow browser window you don't see the work Python...

For the same reason, I thought about naming this step just `C++ tests`... I don't know, suggestions welcome!